cmd/check.go

```diff
@@ -385,42 +366,50 @@ func exportCheckResult(ctx context.Context, d *control.ExportData) {
 }
 
 func displayControlResults(ctx context.Context, executionTree *controlexecute.ExecutionTree) error {
-	outputFormat := viper.GetString(constants.ArgOutput)
-	formatter, _ := controldisplay.GetOutputFormatter(outputFormat)
+	formatter, _, err := parseOutputArg(viper.GetString(constants.ArgOutput))
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
 	formattedReader, err := formatter.Format(ctx, executionTree)
 	if err != nil {
 		return err
 	}
 	_, err = io.Copy(os.Stdout, formattedReader)
-
 	return err
+
+	// outputFormat := viper.GetString(constants.ArgOutput)
+	// formatter, _ := controldisplay.GetOutputFormatter(outputFormat)
+	// formattedReader, err := formatter.Format(ctx, executionTree)
+	// if err != nil {
+	// 	return err
+	// }
+	// _, err = io.Copy(os.Stdout, formattedReader)
+
+	// return err
 }
 
-func exportControlResults(ctx context.Context, executionTree *controlexecute.ExecutionTree, formats []controldisplay.CheckExportTarget) []error {
+func exportControlResults(ctx context.Context, executionTree *controlexecute.ExecutionTree, targets []controldisplay.CheckExportTarget) []error {
 	errors := []error{}
-	for _, format := range formats {
+	for _, target := range targets {
 		if utils.IsContextCancelled(ctx) {
 			// set the error
 			errors = append(errors, ctx.Err())
 			// and skip forward
 			continue
 		}
-		formatter, err := controldisplay.GetExportFormatter(format.Format)
-		if err != nil {
-			errors = append(errors, err)
-			continue
-		}
-		formattedReader, err := formatter.Format(ctx, executionTree)
+
+		formattedReader, err := target.Formatter.Format(ctx, executionTree)
```

Comment:
what is `formattedReader`? som ebetter comments/code naming would make this clearer